src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs

```diff
@@ -53,7 +53,11 @@ internal sealed class BracePairsTaggerProvider : AsynchronousViewTaggerProvider<
 
         protected override ITaggerEventSource CreateEventSource(ITextView? textView, ITextBuffer subjectBuffer)
         {
+            this.ThreadingContext.ThrowIfNotOnUIThread();
+            Contract.ThrowIfNull(textView);
```

Comment:
😕 Why? Nothing here uses the UI thread.

Reply:
This just is so it can fire if we ever break this contract.

Reply:
Would prefer to remove both of these. The first can be removed as both unnecessary and undesirable (only the `ITaggerEventSource.Connect` should be using the main thread), and the second can be removed per https://github.com/dotnet/roslyn/pull/67454#discussion_r1146427239.

Reply:
removed.